lib/DebugTrackWastedRenders.lua

```diff
@@ -0,0 +1,115 @@
+local DebugTrackWastedRenders = {}
+
+-- Tracks total number of renders by type of component
+local totalRenders = 0
+local wastedRendersByType = {}
+local validRendersByType = {}
+
+-- Allows tracking of frames without renders, so we can only report when needed
+local previousTotal = 0
+local framesWithoutUpdate = 0
+-- TODO: Make this configurable somehow without overcomplicating GlobalConfig
+local renderFramesTillSettled = 10
+
+local HeartbeatConnection = nil
+
+--[[
+	Determine if an old and new elements are equal. Used to detect wasted renders
+	when the trackWastedRenders flag is set
+]]
+local function deepEquals(old, new)
+	if old == new then
+		return true
+	end
+
+	if (type(old) == "table" and type(new) == "table") then
+		for key, oldVal in pairs(old) do
+			if not deepEquals(new[key], oldVal) then
+				return false
+			end
+		end
+
+		for key, newVal in pairs(new) do
+			if not deepEquals(newVal, old[key]) then
+				return false
+			end
+		end
+	else
+		return false
+	end
+
+	return true
+end
+
+
+--[[
+	Print accumulated wasted render stats once the number of renders stops increasing
+	for the amount of frames specified in the config
+]]
+local function printStatsWhenSettled()
+	if previousTotal == totalRenders then
+		framesWithoutUpdate = framesWithoutUpdate + 1
+
+		-- If it's been enough frames without new Roact Renders, report results
+		if framesWithoutUpdate == renderFramesTillSettled then
+			-- TODO: More precise/accurate messaging, better spacing
+			-- Print stats on wasted renders
+			print("UI settled. Render stats:")
```

Comment:
Can we change `UI` to `Roact rendering` or something similar? Sometimes Roact gets used for non-UI applications, so this message could be confusing!